docs/rfc/block-registration.md

```diff
@@ -0,0 +1,523 @@
+ - `transforms` - see the [Transforms](/docs/designers-developers/developers/block-api/block-registration.md#transforms-optional) documentation for more details.
+ - `deprecated` - see the [Deprecated Blocks](/docs/designers-developers/developers/block-api/block-deprecation.md) documentation for more details.
+ - `supports` - see the [block supports](/docs/designers-developers/developers/block-api/block-registration.md#supports-optional) documentation page for more details.
+ - `merge` - undocumented as of today. Its role is to handle merging multiple blocks into one.
+ - `getEditWrapperProps` - undocumented as well. Its role is to inject additional props to the block edit's component wrapper.
+ 
+**Example**:
+```js
+wp.blocks.registerBlockType( 'my-block/name', {
+    edit: function() {
+    	// Edit definition goes here.
+    },
+    save: function() {
+    	// Save definition goes here.
+    },
+    supports: {
+        html: false
+    }
+} );
+``` 
+
+## Assets
+
+### `WPDefinedAsset`
+
+The `WPDefinedAsset` type mirrors the shape of params necessary to register scripts and styles using [`wp_register_script`](https://developer.wordpress.org/reference/functions/wp_register_script/) and [`wp_register_style`](https://developer.wordpress.org/reference/functions/wp_register_style/), and then assign these as handles associated with your block using the `script`, `style`, `editor_script`, and `editor_style` block type registration settings. There are three ways you can use them with `block.json` metadata.
+
+#### Handle
+
+First, in WordPress context, it's possible to register a script or style with a regular PHP function call and then assign the handle (`string`) to the corresponding property in `block.json` file.
+
+**Example:**
+
+In `index.php`:
+```php
+wp_register_script(
+	'my-plugin-notice-editor',
+	plugins_url( 'build/editor.js', __FILE__ ),
+	array( 'wp-blocks', 'wp-element', 'wp-i18n' ),
+	'3.0.0'
+);
+```
+
+In `block.json`:
+```json
+{ "editorScript": "my-plugin-notice-editor"  }
+```
+
+#### Object
+
+An asset can also be defined as an object which takes the following shape:
+- `handle` (`string`) - the name of the script.
+- `file` (`string`) - full URL of the script, or path of the script relative to the WordPress root directory.
+- `dependencies` (`string[]` - optional) - an array of registered script handles this script depends on.  Default value: `array()`.
+- `version` (`string`|`bool`|`null` - optional) - string specifying the script version number, if it has one, which is added to the URL as a query string for cache busting purposes. If the version is set to `false`, a version number is automatically added equal to current installed WordPress version. If set to `null`, no version is added. Default value: `false`.
+
+When a block is registered with PHP, it will also automatically register all scripts and styles that are found in the `block.json` file.
+
+**Example:**
+
+In `block.json`:
+```json
+{
+	"editorScript": {
```

Comment:
The whitespace here makes this section difficult to read. This line has a tab and the following have spaces. It would be nice to make sure the indentation nests properly.

Reply:
Fixed  with 42d6c62afc87a73f48e81d65ff224bbcdfe74805.